packages/core/useDevicePixelRatio/index.ts

```diff
@@ -14,24 +15,14 @@ export function useDevicePixelRatio(options: ConfigurableWindow = {}) {
   } = options
 
   const pixelRatio = ref(1)
+  const query = useMediaQuery(() => `(resolution: ${pixelRatio.value}dppx)`, options)
+  let stop = noop
 
   if (window) {
-    let media: MediaQueryList
-    function observe() {
-      pixelRatio.value = window!.devicePixelRatio
-      cleanup()
-      media = window!.matchMedia(`(resolution: ${pixelRatio.value}dppx)`)
-      media.addEventListener('change', observe, { once: true })
-    }
-    function cleanup() {
-      media?.removeEventListener('change', observe)
-    }
-
-    observe()
-    tryOnScopeDispose(cleanup)
+    stop = watchImmediate(query, () => pixelRatio.value = window!.devicePixelRatio)
   }
 
-  return { pixelRatio }
+  return { pixelRatio, stop }
```

Comment:
```suggestion
  return { readonly(pixelRatio), stop }
```

maybe we should readonly this too